gradle-diff-task-dependencies/plugin/src/main/groovy/com/novoda/gradle/diffdependencytask/ext/ExtensionRegister.groovy

```diff
@@ -0,0 +1,27 @@
+package com.novoda.gradle.diffdependencytask.ext
+
+import com.novoda.gradle.diffdependencytask.ConditionalDependency
+import org.gradle.api.Task
+
+import java.util.regex.Pattern
+
+enum ExtensionRegister implements Runnable {
+
+    INSTANCE
+
+    private ExtensionRegister() {
+        // private ctor
+    }
+
+    @Override
+    void run() {
+        registerTaskExtension()
+    }
+
+    private static void registerTaskExtension() {
+        Task.metaClass.ifDiffMatches << { Pattern[] patterns ->
```

Comment:
As mentioned above `Task` as subtype of [`ExtensionAware`](https://docs.gradle.org/current/dsl/org.gradle.api.plugins.ExtensionAware.html) owns an `ExtensionProperties`, therefore you should be able to approach this differently. You could for instance amend all the tasks (including the future ones added) via
```
project.tasks.all { task -> /* blah*/ }
```